src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/mongo/AgoraMongoDao.scala

```diff
@@ -27,7 +27,17 @@ object AgoraMongoDao {
   def MongoDbObjectToEntity(mongoDBObject: DBObject): AgoraEntity = mongoDBObject.toString.parseJson.convertTo[AgoraEntity]
 }
 
-class AgoraMongoDao(collection: MongoCollection) extends AgoraDao {
+/**
+ * The data access object for agora's mongo database. If more than one collection is specified we will query all
+ * collections. However, only a single collection is allowed for doing insert operations.
+ * @param collections The collections to query. In order to insert a single colleciton must be specified.
+ */
+class AgoraMongoDao(collections: Seq[MongoCollection]) extends AgoraDao {
+
+  def getSingleCollection: MongoCollection = {
+    if (collections.size != 1) throw new IllegalArgumentException("Inserts can only be run on a single collection")
```

Comment:
Kind of a deceptive exception message - the method is getSingleCollection and could be used elsewhere than insert.